### ISSUE TITLE
Authenticator not accepting aggregator cert on sample-apiserver

### ISSUE THREAD

**Issue by cheftako** [kubernetes/kubernetes#47194]
**Kubernetes version**:1.7.0

**Environment**: 
- **Cloud provider or hardware configuration**: GCE
-**Other patches**: pull/47094

**What happened**:
Trying to run sample-apiserver with the Aggregator.
"kubectl get ConfigMap extension-apiserver-authentication --namespace=kube-system -o yaml" shows a requestheader-client-ca-file:
However all attempts by the aggregator to the sample-apiserver return a 401.
The logs indicate that this is because of a UnknownAuthorityError.

**What you expected to happen**:
Aggregator would seamlessly proxy requests successfully to the sample-apiserver. 

**How to reproduce it** (as minimally and precisely as possible):
Deploy the sample-apiserver.
Run "kubectl get flunders"
Get back "No resources found."

**Anything else we need to know**:

**Comment by liggitt** [comment]
How did you create the client cert/key for the aggregator? Can you verify the client cert/key the aggregator is using matches the ca in the configmap?

**Comment by cheftako** [comment]
Prospective fix is pull/47196

**Comment by cheftako** [comment]
I created the client cert/key for the aggregator using pull/47094. I connected to the kube-apiserver and verified using openssl that the ca signing the client cert/key on it matched the ca in the configmap. I also instrumented the x509 code on the sample-apiserver to print the CN of the the cert and its issuer in the request header. I validated that they matched the corresponding keys on the kube-apiserver. Lastly I printed the CNs of the CAs in each of the cert pools on the sample-apiserver.
